### PR TITLE
fix(tls_inspector): added AggregateListener support

### PR DESCRIPTION
**Pull request by gunnar-solo**
[envoy.filters.listener.tls_inspector](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/tls_inspector) is an essential [listener_filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/listener_filters) when attempting to use TLS.  For _most_ gateways, we would previously always consider logic on whether or not to add it based on the presence of a user-provided SslConfig field.  For the new [AggregateListener](https://github.com/solo-io/gloo/pull/6558/files#diff-2272a3c8b202d19594dd9a455396b8fd7504a83cdd5eeacc918cecca586f2375), we neglected to add support, originally.  This caused TLS to be unable to function when using the [isolateVirtualHostsBySslConfig](https://docs.solo.io/gloo-edge/latest/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/settings.proto.sk/) feature flag.

related https://github.com/solo-io/gloo/issues/6677

### CODE REVIEW

File: changelog/v1.13.0-beta6/isolate-virtual-hosts-sharing.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: FIX
+    issuelink: https://github.com/solo-io/gloo/issues/6677
+    resolvesIssue: false
+    description: Added support fo the `envoy.filters.listener.tls_inspector` listener_filter when using the gateway.isolateVirtualHostsBySslConfig=true global setting.
+      is provided
```
Comment (sam-heilbron) [inline changelog/v1.13.0-beta6/isolate-virtual-hosts-sharing.yaml:6]: nit: delete?

**Review by sam-heilbron — COMMENTED** [review]
Looks good!

Is it possible to update https://github.com/solo-io/gloo/blob/master/test/e2e/aggregate_listener_test.go to modify a test that would fail without this change and pass with it?

**Comment by gunnar-solo** [comment]
reproduction/test comment:

```bash
# following along with https://docs.solo.io/gloo-edge/latest/guides/security/tls/server_tls/
cat <<EOF | kind create cluster --name kind --image kindest/node:v1.22.4 --config=-
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  extraPortMappings:
  - containerPort: 31500
    hostPort: 31500
    protocol: TCP
  - containerPort: 32500
    hostPort: 32500
    protocol: TCP
EOF

# install new-ish version of edge
helm install -n gloo-system gloo-ee gloo-ee/gloo-ee \
  --create-namespace \
  --version 1.12.3 \
  --set-string license_key=$GLOO_LICENSE_KEY

kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo/v1.11.x/example/petstore/petstore.yaml

# patch settings to enable proxy object creation
kubectl patch settings -n gloo-system default --type='merge' -p '{"spec":{"gateway":{"persistProxySpec":true}}}'

# delete unwanted gateway
kubectl -n gloo-system delete Gateway gateway-proxy

# add test-vs-1
openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
   -keyout tls1.key -out tls1.crt -subj "/CN=petstore.example.com"
kubectl create secret tls upstream-tls-1 --key tls1.key \
   --cert tls1.crt --namespace gloo-system

kubectl -n gloo-system apply -f - <<EOF
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: test-vs-1
  namespace: gloo-system
spec:
  sslConfig:
    secretRef:
      name: upstream-tls-1
      namespace: gloo-system
    sniDomains:
      - petstore.example.com
  virtualHost:
    domains:
      - 'test-domain-1'
    routes:
      - matchers:
          - exact: /sample-route-1
        options:
          prefixRewrite: /api/pets
        routeAction:
          single:
            upstream:
              name: default-petstore-8080
              namespace: gloo-system
EOF

# add test-vs-2
openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
   -keyout tls2.key -out tls2.crt -subj "/CN=petstore2.example.com"
kubectl create secret tls upstream-tls-2 --key tls2.key \
   --cert tls2.crt --namespace gloo-system

kubectl -n gloo-system apply -f - <<EOF
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: test-vs-2
  namespace: gloo-system
spec:
  sslConfig:
    secretRef:
      name: upstream-tls-2
      namespace: gloo-system
    sniDomains:
      - petstore2.example.com
  virtualHost:
    domains:
      - 'test-domain-2'
    routes:
      - matchers:
          - exact: /sample-route-2
        options:
          prefixRewrite: /api/pets
        routeAction:
          single:
            upstream:
              name: default-petstore-8080
              namespace: gloo-system
EOF

# testing
kubectl port-forward -n gloo-system deploy/gateway-proxy 8443

curl -H "Host: test-domain-1" -k --resolve petstore.example.com:8443:127.0.0.1 https://petstore.example.com:8443/sample-route-1
> Handling connection for 8443
> [{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]

curl -H "Host: test-domain-2" -k --resolve petstore2.example.com:8443:127.0.0.1 https://petstore2.example.com:8443/sample-route-2
> Handling connection for 8443
> [{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]

curl -H "Host: test-domain-2" -k --resolve petstore2.example.com:8443:127.0.0.1 https://petstore.example.com:8443/sample-route-2
> curl: (6) Could not resolve host: petstore.example.com
```

**Review by sam-heilbron — APPROVED** [review]

**Comment by gunnar-solo** [comment]
/kick [Step #10 - "test": [Fail] Gateway in memory tcp gateway ssl [It] should work with ssl 
](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=c297fc48-c246-47a5-8a57-18935d8d46dd)

**Comment by nfuden** [comment]
/kick csrf [Details](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=9280f506-81a8-40d1-86e3-e56fd48e6c8b)

**Comment by nfuden** [comment]
/kick data race likely the one thats being bad around set up with syncer

**Pull request merged**